assets/js/account/account-profile-update.js

```diff
@@ -26,7 +26,11 @@ angular.module('storefront.account')
                         fullName: customer.firstName + ' ' + customer.lastName,
                         email: customer.email
                     };
-                    $ctrl.phoneNumber = customer.phoneNumber;
+                    if (customer.contact)
```

Comment:
Probably `customer.phoneNumber` (it is ASPNET user phone number that is used in 2 factor auth) should be shown if exists in favor of one of contact phone numbers.